File: falconeye/characters/validators.py
import re

from django.core.exceptions import ValidationError


def birth_year_validator(birth_year: str) -> None:
    """
    Validate if given birth year is in 19 BBY or 19 ABY format
    """
    if birth_year == "unknown":
        return

    match = re.match(r"(^\d*\.?\d*)\s?(BBY|ABY)", "419BBY", re.I)

    if not match:
        raise ValidationError("Invalid Birth Year")

File: static/ts/validators.ts
import { peopleInitialData } from "./constants";

export const validate = (values: PeopleData): PeopleData => {
  const errors: PeopleData = { ...peopleInitialData };

  if (!values.name) {
    errors.name = "Name is required";
  }

  if (!values.birth_year) {
    errors.birth_year = "Birth Year is required";
  } else if (!/^\d*\.?\d*\s?(ABY|BBY)/.test(values.birth_year)) {
    errors.birth_year = "Birth Year should be in 123 ABY or 123 BBY format";
  }

  const validGender = ["male", "female", "unknown", "n/a"];

  if (!values.gender) {
    errors.gender = "Gender is required";
  } else if (!validGender.indexOf(values.gender)) {
    errors.gender = `Gender should be in  ${validGender.join()}`;
  }

  if (!values.homeworld.name) {
    errors.homeworld.name = "Planet name is required";
  }

  if (!values.homeworld.population) {
    errors.homeworld.population = "Population is required";
  }

  if (!values.homeworld.climate) {
    errors.homeworld.climate = "Climate is required";
  }

  if (!values.homeworld.terrain) {
    errors.homeworld.terrain = "Terrain is required";
  }

  return errors;
};